web/main/models.py

```diff
@@ -2077,41 +2098,61 @@ def contents(self):
         res = ContentNode.objects.filter(**filter_map).exclude(id=self.id)
         return res
 
-    def get_previous_and_next_node_urls(self) -> tuple[Optional[str], Optional[str]]:
+    def get_previous_and_next_nodes(
```

Comment:
When I was adding role awareness to the next/previous arrows I thought that this feature was useful enough that it should more generally return the next and previous _nodes_ , and then turn those into URLs in a separate method, so that's mostly what happened here.

Reply:
Love it.